### PR TITLE
feat: create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by ttacon**
#### Changes Made
 - Create code analysis Github Action

#### Potential Risks
 - Break things

#### Test Plan
 - Open this PR

#### Checklist
- [x] I've increased test coverage - N/A
- [x] Since this is a public repository, I've checked I'm not publishing private data in the code, commit comments, or this PR.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# For most projects, this workflow file will not need changing; you simply need
```
Comment (ghmeier) [inline .github/workflows/codeql-analysis.yml:1]: I wonder if we can add this to CI or something?

**Review by ghmeier — APPROVED** [review]
super neat! i wonder what we'll find!

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (rtertiaer) [reply .github/workflows/codeql-analysis.yml:1]: I'm pretty sure adding this file here, in the `.github/workflows` folder, adds it to Github Actions.

**Review by rtertiaer — APPROVED** [review]
this is super cool stuff!! :rocket:

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (ttacon) [reply .github/workflows/codeql-analysis.yml:1]: Correct, this change makes this live on GitHub Actions.

Comment (ghmeier) [reply .github/workflows/codeql-analysis.yml:1]: Yeah, I just meant that i'd be rad if we could include this somewhere for all repos so we don't have to configure it in github everywhere 🤷‍♀️

**Pull request merged**